dask/array/core.py

```diff
@@ -731,7 +731,7 @@ def store(sources, targets, lock=True, regions=None, compute=True,
         sources_dsk,
         list(core.flatten([e.__dask_keys__() for e in sources]))
     )
-    sources2 = [Array(sources_dsk, e.name, e.chunks, e.dtype) for e in sources]
+    sources2 = [Array(sources_dsk, e.name, e.chunks, meta=e._meta) for e in sources]
```

Comment:
I still think that it would be nice to be able to say `meta=e` here and have this normalized within some sort of normalization function that gets called within the `Array` constructor.

Reply:
Added this to the to-do list at the top, so it isn't forgotten.